### PR TITLE
DT-2644 Use generic service chart for Helm

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl [hunk 1]
```diff
@@ -1,85 +0,0 @@
...
-    value: "{{ .Values.env.OAUTH_ENDPOINT_URL }}"
-
-  - name: PROBATION_ENDPOINT_URL
-    value: "{{ .Values.env.PROBATION_ENDPOINT_URL }}"
-
-  - name: COURT_REGISTER_ENDPOINT_URL
-    value: "{{ .Values.env.COURT_REGISTER_ENDPOINT_URL }}"
-
-  - name: SPRING_PROFILES_ACTIVE
-    value: "logstash"
```
Comment (petergphillips) [inline helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl:23]: can't see this property in `values`

File: helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl [hunk 2]
```diff
@@ -1,85 +0,0 @@
...
-        name: {{ template "app.name" . }}
-        key: APPINSIGHTS_INSTRUMENTATIONKEY
-
-  - name: APPLICATIONINSIGHTS_CONNECTION_STRING
-    value: "InstrumentationKey=$(APPINSIGHTS_INSTRUMENTATIONKEY)"
-
-  - name: APPLICATIONINSIGHTS_CONFIGURATION_FILE
-    value: "{{ .Values.env.APPLICATIONINSIGHTS_CONFIGURATION_FILE }}"
-
-  - name: HMPPS_AUTH_CLIENT_CLIENT_ID
```
Comment (petergphillips) [inline helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl:37]: these two need to be copied from the secret in `values.yml`

File: helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl [hunk 3]
```diff
@@ -1,85 +0,0 @@
...
-  - name: HMPPS_AUTH_CLIENT_CLIENT_SECRET
-    valueFrom:
-      secretKeyRef:
-        name: {{ template "app.name" . }}
-        key: OAUTH_CLIENT_SECRET
-
-  - name: HMPPS_SQS_QUEUES_REGISTERS_QUEUE_ACCESS_KEY_ID
-    valueFrom:
-      secretKeyRef:
-        name: sqs-delius-update-secret
```
Comment (petergphillips) [inline helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl:52]: these queue variables need to be copied too

File: helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl (hunk 2 above)
Comment (karenmillermoj) [reply helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl:37]: APPLICATIONINSIGHTS_CONFIGURATION_FILE set in individual files.  Added HMPPS_AUTH_CLIENT_CLIENT_ID

File: helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl (hunk 1 above)
Comment (karenmillermoj) [reply helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl:23]: added

File: helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl (hunk 3 above)
Comment (karenmillermoj) [reply helm_deploy/hmpps-registers-to-delius-update/templates/_envs.tpl:52]: Added missing queue config

File: helm_deploy/hmpps-registers-to-delius-update/values.yaml [hunk 4]
```diff
@@ -1,5 +1,55 @@
...
+
+  # Pre-existing kubernetes secrets to load as environment variables in the deployment.
+  # namespace_secrets:
+  #   [name of kubernetes secret]:
+  #     [name of environment variable as seen by app]: [key of kubernetes secret to load]
+
+  namespace_secrets:
+    hmpps-registers-to-delius-update:
+      APPINSIGHTS_INSTRUMENTATIONKEY: "APPINSIGHTS_INSTRUMENTATIONKEY"
+      HMPPS_AUTH_CLIENT_CLIENT_ID: "HMPPS_AUTH_CLIENT_CLIENT_ID"
```
Comment (petergphillips) [inline helm_deploy/hmpps-registers-to-delius-update/values.yaml:34]: this should be 
```
      HMPPS_AUTH_CLIENT_CLIENT_ID: "OAUTH_CLIENT_ID"
```
and same with one below

Comment (karenmillermoj) [reply helm_deploy/hmpps-registers-to-delius-update/values.yaml:34]: yes - you're right

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**